### PR TITLE
Check access to product on category listing

### PR DESCRIPTION
**Pull request by bibips**
Products that were not accessible by the user were still visible on the listing

| Questions     | Answers
| ------------- | -------------------------------------------------------
| Description?  | Verifies that the user has the right to access the product
| Type?         | bug fix 
| BC breaks?    |  no
| Deprecations? | no
| Fixed ticket? | Fixes PrestaShop/Prestashop#27601.
| How to test?  | **On Administration**<br/>1. On category  **Clothes**, go to sub-category **Men** and set access only to groups **Customer** <br/>2. Create a product **test** with only the category **Men**<br/>**On store**<br/>1. Go to the category **Clothes**<br/>2. Sub-category **Men** isn't display and you can't see the produt **test**

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/ps_facetedsearch/612)

### CODE REVIEW

**Review by PierreRambaud — CHANGES_REQUESTED** [review]
Thanks for your contribution, but I think you didn't test it at all, as the Group class doesn't exist in this scope, same for FrontController

**Review by PierreRambaud — DISMISSED** [review]

**Comment by bibips** [comment]
@PierreRambaud Thanks for your review

**Comment by PierreRambaud** [comment]
> @PierreRambaud Thanks for your review

Let's wait for other maintainer to review this PR and put it in QA :)

**Review by kpodemski — CHANGES_REQUESTED** [review]
Could you add the test based on the group that doesn't have access to the specific category? IMHO right now, you're testing a scenario that is always true.

**Comment by bibips** [comment]
@kpodemski I just added a commit with more tests

**Review by PierreRambaud — APPROVED** [review]

**Review by Progi1984 — APPROVED** [review]

**Review by florine2623 — APPROVED** [review]
Hello @bibips ,

Thanks for the PR !

Initial case is looking good to me.

https://user-images.githubusercontent.com/16019289/157853401-b33b43fb-8ca1-416d-9026-5f5a0bb2c009.mov

Tested on existing customer groups, newly created customer groups.
Tested on Disabled/Deleted Customer groups.
Checked case when Category is disabled/deleted.
Checked product in FO when product is in multiple categories.

It is QA ✅ 
Thanks!

**Comment by Progi1984** [comment]
Thanks @florine2623  & @bibips

**Pull request merged**
